src/middlewared/middlewared/plugins/truecommand/connection.py

```diff
@@ -0,0 +1,33 @@
+import aiohttp
+import async_timeout
+import asyncio
+import os
+import json
+
+
+class TruecommandAPIMixin:
+
+    PORTAL_URI = 'https://portal.ixsystems.com/api'
+
+    async def _post_call(self, options=None, payload=None):
+        options = options or {}
+        timeout = options.get('timeout', 15)
+        response = {'error': None, 'response': {}}
+        try:
+            with async_timeout.timeout(timeout):
+                async with aiohttp.ClientSession(
+                    raise_for_status=True
+                ) as session:
+                    req = await session.post(
+                        self.PORTAL_URI,
+                        data=json.dumps(payload or {}),
+                        headers={'Content-type': 'application/json'},
+                        proxy=os.environ.get('http_proxy'),
```

Comment:
I think there is possible to have a https_proxy as well?

Reply:
```
aiohttp supports plain HTTP proxies and HTTP proxies that can be upgraded to HTTPS via the HTTP CONNECT method. aiohttp does not support proxies that must be connected to via https://.
```
I think we should be safe, or do you think that is not the case ?

Reply:
Use `aiohttp.ClientSession(trust_env=True)` instead of this